agent/src/com/cloud/agent/Agent.java

```diff
@@ -265,6 +273,8 @@ public void start() {
                 s_logger.info("Attempted to connect to the server, but received an unexpected exception, trying again...");
             }
         }
+        _shell.updateConnectedHost();
```

Comment:
@nvazquez to be on safer/deterministic side, can this method accept the `host` String and set that instead of `hostToConnect`?

Reply:
@rhtyd I decided to take that path as when agent is launched, the next host is set on constructor and then from `start()` method we couldn't know which host the agent is trying to connect to unless we do the same as in AgentShell with `hostToConnect` but on Agent class. Which approach do you think it would be best?